frontend/frontend_base.h

```diff
@@ -583,29 +583,16 @@ template <typename FrontendType> struct GenericFrontend
             connect_port(ctx, net, iobuf, ctx->id("I"));
         } else if (dir == PORT_INOUT) {
             iobuf->type = ctx->id("$nextpnr_iobuf");
-            iobuf->addInput(ctx->id("I"));
-            iobuf->addOutput(ctx->id("O"));
-            // Need to bifurcate the net to avoid multiple drivers and split
-            // the input/output parts of an inout
-            // Create a new net connecting only the current net's driver and the IOBUF input
-            // Then use the IOBUF output to drive all of the current net's users
-            NetInfo *split_iobuf_i = ctx->createNet(unique_name("", "$" + name + "$iobuf_i", true));
-            auto drv = net->driver;
-            if (drv.cell != nullptr) {
-                disconnect_port(ctx, drv.cell, drv.port);
-                drv.cell->ports[drv.port].net = nullptr;
-                connect_port(ctx, split_iobuf_i, drv.cell, drv.port);
-            }
-            connect_port(ctx, split_iobuf_i, iobuf, ctx->id("I"));
-            NPNR_ASSERT(net->driver.cell == nullptr);
-            connect_port(ctx, net, iobuf, ctx->id("O"));
+            iobuf->addInout(ctx->id("IO"));
```

Comment:
This will break other arches; I do think the whole IO buffer stuff is not the nicest part of nextpnr and is somewhat a legacy of the decision to support inferring IO buffers (including complex tristate stuff) in the ice40 and ecp5 arches. If you have any suggestions on improving this; I'd be interested to know (one option is an arch flag to disable the creation of the buffers altogether for arches that don't need them; then they can just look at `ctx->ports`)

Reply:
The problem as I see it right now is that nextpnr doesn't have a place to attach `port` properties, e.g. `PACKAGE_PIN`, `IOSTANDARD` (or equivalent).  The iobuf cells that are created here are a reasonable place for now.

I'll think on this today.  In general, we want the placer to be able to choose where IO ports and their buffers are in the fabric, and that does mean a placable object must exist.  I think that inout top level ports should be represented by 1 wire that connects to the tristate buffer and local pad objects (e.g. pulls).

As a short term fix, what do you think about gating the `$nextpnr_iobuf` behavior (e.g. 1 port named "IO" versus two ports named "I" and "O") behind an argument to `GenericFrontend`?

Reply:
> As a short term fix, what do you think about gating the $nextpnr_iobuf behavior (e.g. 1 port named "IO" versus two ports named "I" and "O") behind an argument to GenericFrontend?

SGTM

Reply:
This was fixed in #579